File: android/app/src/main/java/com/example/workflowupdater/ui/main/WorkCard.kt
package com.example.workflowupdater.ui.main

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material.icons.filled.SquareFoot
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import com.example.workflowupdater.data.SheetConfig
import com.example.workflowupdater.data.WorkItem
import com.example.workflowupdater.ui.common.statusTone

@Composable
fun WorkCard(work: WorkItem, onClick: () -> Unit, modifier: Modifier = Modifier) {
  Card(
    onClick = onClick,
    modifier = modifier.fillMaxWidth(),
    shape = RoundedCornerShape(18.dp),
    colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceContainer),
  ) {
    Column(modifier = Modifier.padding(16.dp)) {
      Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.SpaceBetween, modifier = Modifier.fillMaxWidth()) {
        Text(
          text = work.fileNumber.ifBlank { "No file number" },
          style = MaterialTheme.typography.labelMedium,
          color = MaterialTheme.colorScheme.onSurfaceVariant,
          maxLines = 1,
          overflow = TextOverflow.Ellipsis,
          modifier = Modifier.weight(1f, fill = false),
        )
        Spacer(Modifier.width(8.dp))
        StatusBadge(status = work.status, code = work.statusCode)
      }

      Spacer(Modifier.height(6.dp))

      Text(
        text = work.workName,
        style = MaterialTheme.typography.titleMedium,
        color = MaterialTheme.colorScheme.onSurface,
        maxLines = 2,
        overflow = TextOverflow.Ellipsis,
      )

      Spacer(Modifier.height(10.dp))

      Row(verticalAlignment = Alignment.CenterVertically) {
        Icon(Icons.Filled.LocationOn, contentDescription = null, tint = MaterialTheme.colorScheme.onSurfaceVariant, modifier = Modifier.size(14.dp))
        Spacer(Modifier.width(4.dp))
        Text(
          text = listOf(work.lac, work.district).filter { it.isNotBlank() }.joinToString(" \u2022 ").ifBlank { "Location not set" },
          style = MaterialTheme.typography.bodySmall,
          color = MaterialTheme.colorScheme.onSurfaceVariant,
          maxLines = 1,
          overflow = TextOverflow.Ellipsis,
          modifier = Modifier.weight(1f),
        )
      }

      if (work.floors.isNotBlank() || work.area.isNotBlank()) {
        Spacer(Modifier.height(4.dp))
        Row(verticalAlignment = Alignment.CenterVertically) {
          Icon(Icons.Filled.SquareFoot, contentDescription = null, tint = MaterialTheme.colorScheme.onSurfaceVariant, modifier = Modifier.size(14.dp))
          Spacer(Modifier.width(4.dp))
          Text(
            text = listOfNotNull(
                work.floors.ifBlank { null }?.let { "$it floors" },
                work.area.ifBlank { null }?.let { "$it m\u00b2" },
              )
              .joinToString("  \u2022  "),
            style = MaterialTheme.typography.bodySmall,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
          )
        }
      }

      Spacer(Modifier.height(10.dp))

      Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
        MiniStatusPill(label = "AS", value = work.asStatus)
        MiniStatusPill(label = "AR", value = work.arStatus)
        MiniStatusPill(label = "SR", value = work.srStatus)
      }

      if (work.remarks.isNotBlank()) {
        Spacer(Modifier.height(10.dp))
        Text(
          text = work.remarks,
          style = MaterialTheme.typography.bodySmall,
          color = MaterialTheme.colorScheme.onSurfaceVariant,
          maxLines = 2,
          overflow = TextOverflow.Ellipsis,
        )
      }

      Spacer(Modifier.height(10.dp))

      Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.SpaceBetween, modifier = Modifier.fillMaxWidth()) {
        Text(
          text = "View full details",
          style = MaterialTheme.typography.labelMedium,
          color = MaterialTheme.colorScheme.primary,
        )
        Icon(Icons.Filled.ChevronRight, contentDescription = null, tint = MaterialTheme.colorScheme.primary, modifier = Modifier.size(18.dp))
      }
    }
  }
}

@Composable
private fun StatusBadge(status: String, code: String) {
  val tone = statusTone(code)
  val label = SheetConfig.STATUS_SHORT_LABELS[code] ?: status
  Row(
    modifier =
      Modifier.background(color = tone.background, shape = RoundedCornerShape(20.dp))
        .padding(horizontal = 10.dp, vertical = 4.dp),
  ) {
    Text(text = label, style = MaterialTheme.typography.labelSmall, color = tone.foreground, maxLines = 1)
  }
}

@Composable
private fun MiniStatusPill(label: String, value: String) {
  val display = value.ifBlank { "\u2014" }
  Row(
    modifier =
      Modifier.background(color = MaterialTheme.colorScheme.surfaceContainerHigh, shape = RoundedCornerShape(8.dp))
        .padding(horizontal = 8.dp, vertical = 4.dp),
    verticalAlignment = Alignment.CenterVertically,
  ) {
    Text(text = "$label:", style = MaterialTheme.typography.labelSmall, color = MaterialTheme.colorScheme.onSurfaceVariant)
    Spacer(Modifier.width(4.dp))
    Text(
      text = display,
      style = MaterialTheme.typography.labelSmall,
      color = MaterialTheme.colorScheme.onSurface,
      maxLines = 1,
      overflow = TextOverflow.Ellipsis,
    )
  }
}

File: docs/works/js/ui/workCard.js
/** One work in the list. Ported from android/.../ui/main/WorkCard.kt. */

import { STATUS_SHORT_LABELS } from '../config.js';
import { el } from './dom.js';
import { Icons } from './icons.js';
import { statusTone } from './statusTone.js';

export function workCard(work, onClick) {
  const card = el('article', {
    className: 'work-card',
    attrs: { tabindex: '0', role: 'button', 'aria-label': `${work.workName}. View full details` },
    on: {
      click: () => onClick(work.rowNum),
      keydown: (event) => {
        if (event.key === 'Enter' || event.key === ' ') {
          event.preventDefault();
          onClick(work.rowNum);
        }
      },
    },
  });

  card.append(
    el('div', { className: 'work-card-top' }, [
      el('span', { className: 'work-file-number', text: work.fileNumber || 'No file number' }),
      statusBadge(work.status, work.statusCode),
    ]),
    el('h3', { className: 'work-title', text: work.workName }),
    metaRow(Icons.locationOn(), [work.lac, work.district].filter(Boolean).join(' • ') || 'Location not set'),
  );

  if (work.floors || work.area) {
    const parts = [];
    if (work.floors) parts.push(`${work.floors} floors`);
    if (work.area) parts.push(`${work.area} m²`);
    card.append(metaRow(Icons.squareFoot(), parts.join('  •  ')));
  }

  card.append(
    el('div', { className: 'mini-pill-row' }, [
      miniPill('AS', work.asStatus),
      miniPill('AR', work.arStatus),
      miniPill('SR', work.srStatus),
    ]),
  );

  if (work.remarks) {
    card.append(el('p', { className: 'work-remarks', text: work.remarks }));
  }

  card.append(
    el('div', { className: 'work-card-footer' }, [
      el('span', { text: 'View full details' }),
      el('span', { className: 'chevron', html: Icons.chevronRight() }),
    ]),
  );

  return card;
}

function statusBadge(status, code) {
  return el('span', {
    className: 'status-badge',
    text: STATUS_SHORT_LABELS[code] || status,
    dataset: { tone: statusTone(code) },
  });
}

function metaRow(icon, text) {
  return el('div', { className: 'work-meta' }, [el('span', { className: 'meta-icon', html: icon }), el('span', { text })]);
}

function miniPill(label, value) {
  return el('span', { className: 'mini-pill' }, [
    el('span', { className: 'mini-pill-label', text: `${label}:` }),
    el('span', { className: 'mini-pill-value', text: value || '—' }),
  ]);
}
